prow/config/secret/agent.go

```diff
@@ -39,37 +51,35 @@ type Agent struct {
 // Start creates goroutines to monitor the files that contain the secret value.
 // Additionally, Start wraps the current standard logger formatter with a
 // censoring formatter that removes secret occurrences from the logs.
-func (a *Agent) Start(paths []string) error {
+func Start(paths []string) error {
 	secretsMap, err := LoadSecrets(paths)
 	if err != nil {
 		return err
 	}
 
-	a.secretsMap = secretsMap
-	a.ReloadingCensorer = secretutil.NewCensorer()
-	a.refreshCensorer()
-
-	logrus.SetFormatter(logrusutil.NewFormatterWithCensor(logrus.StandardLogger().Formatter, a.ReloadingCensorer))
+	for key, value := range secretsMap {
+		agent.setSecret(key, value)
```

Comment:
Technically this ends up with us calling `refreshCensorer()` N times instead of once, as it was before, but I am not sure we can do better when multiple `Start()` calls may be happening at once and interleaved